utility/src/test/java/org/oppia/util/logging/EventBundleCreatorTest.kt

```diff
@@ -0,0 +1,197 @@
+import org.robolectric.annotation.Config
+import javax.inject.Inject
+import javax.inject.Singleton
+
+const val TEST_TIMESTAMP = 1556094120000
+const val TEST_TOPIC_ID = "test_topicId"
+const val TEST_STORY_ID = "test_storyId"
+const val TEST_EXPLORATION_ID = "test_explorationId"
+const val TEST_QUESTION_ID = "test_questionId"
+
+@RunWith(AndroidJUnit4::class)
+@Config(manifest = Config.NONE)
+class EventBundleCreatorTest {
+  @Inject
+  lateinit var fakeEventLogger: FakeEventLogger
+
+  private val eventLogExplorationContext = EventLog.newBuilder()
+    .setActionName(EventLog.EventAction.EVENT_ACTION_UNSPECIFIED)
+    .setContext(
+      EventLog.Context.newBuilder()
+        .setExplorationContext(
+          EventLog.ExplorationContext.newBuilder()
+            .setTopicId(TEST_TOPIC_ID)
+            .setExplorationId(TEST_EXPLORATION_ID)
+            .setStoryId(TEST_STORY_ID)
+            .build()
+        )
+        .build()
+    )
+    .setTimestamp(TEST_TIMESTAMP)
+    .setPriority(EventLog.Priority.ESSENTIAL)
+    .build()
+
+  private val eventLogQuestionContext = EventLog.newBuilder()
+    .setActionName(EventLog.EventAction.EVENT_ACTION_UNSPECIFIED)
+    .setContext(
+      EventLog.Context.newBuilder()
+        .setQuestionContext(
+          EventLog.QuestionContext.newBuilder()
+            .setTopicId(TEST_TOPIC_ID)
+            .setQuestionId(TEST_QUESTION_ID)
+            .build()
+        )
+        .build()
+    )
+    .setTimestamp(TEST_TIMESTAMP)
+    .setPriority(EventLog.Priority.ESSENTIAL)
+    .build()
+
+  private val eventLogTopicContext = EventLog.newBuilder()
+    .setActionName(EventLog.EventAction.EVENT_ACTION_UNSPECIFIED)
+    .setContext(
+      EventLog.Context.newBuilder()
+        .setTopicContext(
+          EventLog.TopicContext.newBuilder()
+            .setTopicId(TEST_TOPIC_ID)
+            .build()
+        )
+        .build()
+    )
+    .setTimestamp(TEST_TIMESTAMP)
+    .setPriority(EventLog.Priority.ESSENTIAL)
+    .build()
+
+  private val eventLogStoryContext = EventLog.newBuilder()
+    .setActionName(EventLog.EventAction.EVENT_ACTION_UNSPECIFIED)
+    .setContext(
+      EventLog.Context.newBuilder()
+        .setStoryContext(
+          EventLog.StoryContext.newBuilder()
+            .setTopicId(TEST_TOPIC_ID)
+            .setStoryId(TEST_STORY_ID)
+            .build()
+        )
+        .build()
+    )
+    .setTimestamp(TEST_TIMESTAMP)
+    .setPriority(EventLog.Priority.ESSENTIAL)
+    .build()
+
+  private val eventLogNoContext = EventLog.newBuilder()
+    .setTimestamp(TEST_TIMESTAMP)
+    .setPriority(EventLog.Priority.ESSENTIAL)
+    .build()
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testBundleCreation_logEvent_withExplorationContext_isSuccessful() {
+    fakeEventLogger
+      .logEvent(ApplicationProvider.getApplicationContext(), eventLogExplorationContext)
+    val eventBundle = fakeEventLogger.getMostRecentEventBundle()
+
```

Comment:
You do not need to create a fake logger. You can directly create an instance of the EventBundleCreator and call its functions. Same comment here and below for all tests. These tests shouldn't have anything to do with the event logger

Reply:
Yes, makes sense. Modified accordingly.